### ISSUE TITLE
formatMessage.setup() mutates options

### ISSUE THREAD

**Issue by epeli** [format-message/format-message#204]
Hi, I'm using format-message in a Redux app where I deep freeze the state in development. My translation strings are in this deep freezed Redux state. When I pass the translations object to the `formatMessage.setup({language: language, translations: translations})` I get this error

    Unhandled JS Exception: TypeError: Cannot add property format, object is not extensible

from

https://github.com/format-message/format-message/blob/9f0677e019e13ff1d852ba34823e28413b95de77/packages/format-message/index.js#L88

I can workaround this by deep cloning the object using lodash `deepClone`: `formatMessage.setup({language: language, translations: deepClone(translations)})`

I'm using format-message 6.1.0.

**Comment by vanwagonet** [comment]
Yeah, it'd be better if format-message didn't mess with what gets passed in. We should rework that code to be immutable object friendly.
